app/src/main/java/org/simple/clinic/drugs/selectionv2/dosage/DosagePickerSheetController.kt

```diff
@@ -49,4 +55,38 @@ class DosagePickerSheetController @Inject constructor(
         }
         .map { dosages -> { ui: Ui -> ui.populateDosageList(dosages) } }
   }
+
+  private fun handleDosageType(): ObservableTransformer<UiEvent, UiEvent> {
```

Comment:
I tried removing the `DosageSelected` and `None` UiEvents and just using the `DosageOption` directly in the streams. This is what I came up with:

```kotlin
val dosageSelected = events
        .ofType<DosageItemClicked>()
        .map { it.dosage }
        .ofType<DosageOption.Dosage>()

val noneSelected =events
        .ofType<DosageItemClicked>()
        .map { it.dosage }
        .ofType<DosageOption.None>()
```

This removes the need for those two event types and the need to further transform the event stream without changing the rest of the streams at all.

Reply:
The other reason why I wanted to keep it as a separate transformer was to use `when`. To make sure I am handling all the cases for `DosageOption`.

Reply:
Is there any issue in using `ObservableTransformer` and merging it with the original stream? 
It is possible that I may not be understanding it completely.

Reply:
The issue is that this transformation is mostly unnecessary since we are not combining any events to create new events. Just mapping one type of event to another type which is, for all intents and purposes, the same event.

Reply:
The `when` clause is also used only in the transformer itself, and not outside it, which makes it unnecessary.

Reply:
Yea, but it does ensure that we are handling all the possible cases for `DosageOption`.
It will throw an error if you add a new class to `DosageOption`.
I would like to keep it this way if it is not causing any issues.

Reply:
It's not causing any issues, but my opinion is that it is unnecessary work that we are doing.

Let's leave this in for now, and revisit this later.